lib/ws/ws.js

```diff
@@ -171,6 +171,12 @@ BeamSocket.prototype._getNextReconnectInterval = function () {
 BeamSocket.prototype._handleClose = function () {
     clearTimeout(this._pingTimeoutHandle);
     this._pingTimeoutHandle = null;
+
+    this.ws.removeAllListeners();
```

Comment:
This will error when used in the DOM, [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) doesn't have a `removeAllListeners` method